### DISCUSSION TITLE
Binning motion data over time

### DISCUSSION THREAD

**Discussion by ktonsfeldt** [bonsai-rx/bonsai#1757] [Q&A]
Hi,

I am using Bonsai to record measure motion data (straight out of Exercise 12 in the tutorial) over weeks at a time. I would like to have Bonsai sum the data over, say, 6 minute intervals, and only write that to CSV. I suspect it can be solved with the timer or takeuntil functions, but can't quite figure out how to get it to sum. 

Thanks!

**Comment by bruno-f-cruz** [comment]
There are several ways to achieve this. I would look into any of the [`Buffer`](https://bonsai-rx.org/docs/api/Bonsai.Reactive.BufferTime.html) operators, [`Window`](https://bonsai-rx.org/docs/api/Bonsai.Reactive.WindowTime.html) or [`CreateObservable` + `Switch`](https://bonsai-rx.org/docs/api/Bonsai.Reactive.Switch.html)
